### DISCUSSION TITLE
Database cleanup & updates

### DISCUSSION THREAD

**Discussion by CGoodwin90** [uselagoon/lagoon#3277] [Roadmap]
The current database schema contains a number of legacy tables and inconsistent rows which would benefit from a cleanup.
After internal discussion we decided to simultaneously update how the DB is built and how schema migrations are managed.

Possible options for running DB schema migrations and versioning for Core:

- [Knex.js](https://knexjs.org/)
- [LiquidBase](https://www.liquibase.org/)
- [Golang Migrate](https://github.com/golang-migrate/migrate)

Initial discussion and investigation has Knex.js as the most suitable option - it's already being utilized throughtout the Lagoon API & the Knex Migration CLI should require little integrration work.

**Comment by tobybellwood** [comment]
Knex looks promising - the ability to handle versioned migrations, save migration state, and selective rollback are all large positives for me.

We should look into how their migration cli operates to figure how best to drive this out of the API image as well, to enhance flexibility.

**Comment by tobybellwood** [comment]
implemented knex in #3318
